### PR TITLE
Refactor Veff.py

### PR DESCRIPTION
**Pull request by fschlueter**
I will comment about a few changes I am proposing in the code. I might have fixed a bug. Unfortunately, the changes harder to track due to the removal of an indentation. If you are fine with these changes more can be done here.

### CODE REVIEW

File: NuRadioMC/utilities/Veff.py [hunk 1]
```diff
@@ -201,33 +224,30 @@ def get_Veff_Aeff_single(filename, trigger_names, trigger_names_dict, trigger_co
...
 
     # calculate effective
-    thetamin = 0
-    thetamax = np.pi
-    phimin = 0
-    phimax = 2 * np.pi
-    if('thetamin' in fin.attrs):
-        thetamin = fin.attrs['thetamin']
-    if('thetamax' in fin.attrs):
-        thetamax = fin.attrs['thetamax']
```
Comment (fschlueter) [inline NuRadioMC/utilities/Veff.py:230]: Do we need those defaults? The arguments should always be in a file!?

File: NuRadioMC/utilities/Veff.py [hunk 2]
```diff
@@ -301,155 +323,160 @@ def get_Veff_Aeff_single(filename, trigger_names, trigger_names_dict, trigger_co
...
-                e = get_efficiency(max_amplitudes / Vrms)  # we calculated the maximum amplitudes for all gids, now we select only those that triggered
-                Veff = volume_proj_area * np.sum(weights[triggered] * e) / n_events
-                Vefferror = 0
-                if(np.sum(weights[triggered]) > 0):
-                    Vefferror = Veff / np.sum(weights[triggered] * e) ** 0.5
-                FC_low, FC_high = FC_limits(np.sum(weights[triggered] * e))
-                Veff_low = volume_proj_area * FC_low / n_events
-                Veff_high = volume_proj_area * FC_high / n_events
-
-            out[veff_aeff][trigger_name] = [Veff, Vefferror, np.sum(weights[triggered]), Veff_low, Veff_high]
```
Comment (fschlueter) [inline NuRadioMC/utilities/Veff.py:448]: Was this line correct? The 3 item in the list should have been `np.sum(weights[triggered]) * e` correct? I fixed that

**Comment by fschlueter** [comment]
@shallmann, maybe you can have a look?

**Comment by fschlueter** [comment]
Ping

**Review by cg-laser — APPROVED** [review]
I think it is ok, but it is really difficult to review because you changed the indentation, so I take your word for it that you made sure that it works as before. In general, I'm not sure if these minor cosmetic tweaks are really necessary as properly reviewing them is difficult ;-)

File: NuRadioMC/utilities/Veff.py (hunk 2 above)
Comment (fschlueter) [reply NuRadioMC/utilities/Veff.py:448]: Hi @cg-laser, I think this line 448 had a bug: For the calculation of `Veff_low` and `Veff_high` a `count` of `np.sum(weights[triggered] * e` (L 444) was used. However in the output only `np.sum(weights[triggered])` is stored for the count

**Comment by fschlueter** [comment]
> I think it is ok, but it is really difficult to review because you changed the indentation, so I take your word for it that you made sure that it works as before. In general, I'm not sure if these minor cosmetic tweaks are really necessary as properly reviewing them is difficult ;-)

Yes I agree, that makes it very hard to check the changes. Sorry.

**Comment by fschlueter** [comment]
> oh, and which bug do you think you fixed?

I answered in the comment to the specific code line

File: NuRadioMC/utilities/Veff.py (hunk 2 above)
Comment (cg-laser) [reply NuRadioMC/utilities/Veff.py:448]: ah ok, got it, this was in the "efficiency" if statement. I implemented this ages ago and I'm not sure if anyone has been using this functionality in the last years ;-) But good to have this consistent now :-)

**Review by cg-laser — APPROVED** [review]

**Pull request merged**